CONTRIBUTING.md

```diff
@@ -368,54 +447,101 @@ The staging workflow is used for all main branches, `master` and `release-YY.MM`
 
-All the review template samples provided in this section are generic and meant as examples. Their usage is optional and the reviewer is free to adapt them to their liking.
+Only changes to supported releases may be accepted.
+The oldest supported release (`YYMM`) can be found using
+```
+nix-instantiate --eval -A lib.trivial.oldestSupportedRelease
+```
 
-To get more information about how to review specific parts of Nixpkgs, refer to the documents linked to in the [overview section](#overview).
+The release branches should generally not receive any breaking changes, both for the Nix expressions and derivations.
+So these changes are acceptable to backport:
+- New packages, modules and functions
+- Security fixes
+- Package version updates
+  - Patch versions with fixes
+  - Minor versions with new functionality, but no breaking changes
 
-If you consider having enough knowledge and experience in a topic and would like to be a long-term reviewer for related submissions, please contact the current reviewers for that topic. They will give you information about the reviewing process. The main reviewers for a topic can be hard to find as there is no list, but checking past pull requests to see who reviewed or git-blaming the code to see who committed to that topic can give some hints.
+In addition, major package version updates with breaking changes are also acceptable for:
+- Services that would fail without up-to-date client software, such as `spotify`, `steam`, and `discord`
+- Security critical applications, such as `firefox` and `chromium`
 
-Container system, boot system and library changes are some examples of the pull requests fitting this category.
+### Changes causing mass rebuilds
+[mass-rebuild]: #changes-causing-mass-rebuilds
 
-## Merging pull requests
+Which changes cause mass rebuilds is not formally defined.
+In order to help the decision, CI automatically assigns [`rebuild` labels](https://github.com/NixOS/nixpkgs/labels?q=rebuild) to pull requests based on the number of packages they cause rebuilds for.
+As a rule of thumb, if the number of rebuilds is **over 500**, it can be considered a mass rebuild.
+To get a sense for what changes are considered mass rebuilds, see [previously merged pull requests to the staging branches](https://github.com/NixOS/nixpkgs/issues?q=base%3Astaging+-base%3Astaging-next+is%3Amerged).
 
-The *Nixpkgs committers* are people who have been given
-permission to merge.
+## Commit conventions
+[commit-conventions]: #commit-conventions
 
-It is possible for community members that have enough knowledge and experience on a special topic to contribute by merging pull requests.
+- When changing the bootloader installation process, extra care must be taken. Grub installations cannot be rolled back, hence changes may break people’s installations forever. For any non-trivial change to the bootloader please file a PR asking for review, especially from \@edolstra.
```

Comment:
I think it's a bit weird to have this point in the "Commit conventions" section.

Reply:
Oh yeah, I initially wasn't sure where it would fit, but now I'm realizing that `nixos/README.md` would be great: f22e4a3106260cdcc48f41f8f4cb483d9520a39e (of course, that document also desperately needs more content and a cleanup, but that's future work)